components/action-mgt/org.wso2.carbon.identity.action.management/src/test/java/org/wso2/carbon/identity/action/management/dao/ActionManagementDAOImplTest.java

```diff
@@ -0,0 +1,586 @@
+    private static final int TENANT_ID = 2;
+    private Action action;
+
+    @BeforeClass
+    public void setUpClass() throws Exception {
+
+        daoImpl = new ActionManagementDAOImpl();
+        initiateH2Database(getFilePath());
+    }
+
+    @BeforeMethod
+    public void setUp() throws SQLException, SecretManagementException {
+
+        identityTenantUtil = mockStatic(IdentityTenantUtil.class);
+        identityDatabaseUtil = mockStatic(IdentityDatabaseUtil.class);
+        SecretManagerImpl secretManager = mock(SecretManagerImpl.class);
+        SecretType secretType = mock(SecretType.class);
+        ActionMgtServiceComponentHolder.getInstance().setSecretManager(secretManager);
+        mockDBConnection();
+        identityTenantUtil.when(()-> IdentityTenantUtil.getTenantId(anyString())).thenReturn(TENANT_ID);
+        when(secretType.getId()).thenReturn("secretId");
+        when(secretManager.getSecretType(any())).thenReturn(secretType);
+    }
+
+    @AfterMethod
+    public void tearDown() {
+
+        identityTenantUtil.close();
+        identityDatabaseUtil.close();
+    }
+
+    @AfterClass
+    public void wrapUp() throws Exception {
+
+        closeH2Database();
+    }
+
+    @Test(priority = 1)
+    public void testAddAction() throws ActionMgtException {
+
+        Action creatingAction = buildMockActionWithBasicAuth(
+                "PreIssueAccessToken",
+                "To configure PreIssueAccessToken",
+                "https://example.com",
+                "admin",
+                "admin");
+        action = daoImpl.addAction(PRE_ISSUE_ACCESS_TOKEN, creatingAction.getId(), creatingAction, TENANT_ID);
+        Assert.assertEquals(creatingAction.getId(), action.getId());
+        Assert.assertEquals(creatingAction.getName(), action.getName());
+        Assert.assertEquals(creatingAction.getDescription(), action.getDescription());
+        Assert.assertEquals(creatingAction.getType(), action.getType());
+        Assert.assertEquals(creatingAction.getStatus(), action.getStatus());
+        Assert.assertEquals(creatingAction.getEndpoint().getUri(), action.getEndpoint().getUri());
+        Assert.assertEquals(creatingAction.getEndpoint().getAuthentication().getType(),
+                action.getEndpoint().getAuthentication().getType());
+    }
+
+    @Test(priority = 2)
+    public void testAddActionWithoutName() {
+
+        Action action = buildMockActionWithBasicAuth(
+                null,
+                "To configure PreIssueAccessToken",
+                "https://example.com",
+                "admin",
+                "admin");
+        try {
+            this.action = daoImpl.addAction(PRE_ISSUE_ACCESS_TOKEN, action.getId(), action, TENANT_ID);
+
+        } catch (Exception e) {
+            Assert.assertEquals(ActionMgtConstants.ErrorMessages.ERROR_WHILE_ADDING_ACTION.getMessage(),
+                    e.getMessage());
+        }
```

Comment:
You can expect this exception to be thrown with @Test annotation and even the message can be defined there

Reply:
Updated in